### PR TITLE
feat: continue chat via existing file upload

### PR DESCRIPTION
**Pull request by AidanB446**
In the issues for this repo, a feature to continue a previous chat was discussed. This implements that feature.

Asks if they would like to continue chat, then shows them possible chat files to continue. Then gets the names from the chat files without user input and continues the conversation appending to that choosen file.

### CODE REVIEW

File: chat.js [hunk 1]
```diff
@@ -12,35 +13,95 @@ const chatLog = [];
...
+let fileChoosen = '';
 
-rl.question('Enter first person name: ', (name1) => {
-    first_person = name1;
-    rl.question('Enter second person name: ', (name2) => {
-        second_person = name2;
-        current_person = first_person;
-        chat();
-    });
+rl.question('Continue from previous chat? y/n :', (cont) => {
```
Comment (fyodorio) [inline chat.js]: **suggestion:** tests show that currently leaving the answer empty defaults to `n`. That's perfect, that's how it should behave. I just think we need to highlight that the way it's usually done in CLI interfaces — by uppercasing the default options. So here's what I propose:
```suggestion
rl.question('Continue from previous chat? y/N :', (cont) => {
```

File: chat.js [hunk 2]
```diff
@@ -12,35 +13,95 @@ const chatLog = [];
...
+	if (cont === "y") {
```
Comment (fyodorio) [inline chat.js]: **suggestion:** we need to make that case-insensitive, as my first habit personally is to answer with <kbd>Shift</kbd>+<kbd>Y</kbd> 😅.

So either

```suggestion
	if (cont.toLowerCase() === "y") {
```

Or
```suggestion
	if (cont === "y" || cont === "Y") {
```

should work

File: chat.js [hunk 3]
```diff
@@ -12,35 +13,95 @@ const chatLog = [];
...
+				second_person = name2;
+				current_person = first_person;
+				chat();
+			});
+		});
+
+	}
 });
 
+function chatCont() {
```
Comment (fyodorio) [inline chat.js:69]: **issue:** looks like we're missing a line break when we merge updates to an existing chat eventually

Screenshot 2024-03-15 at 10 27 16

**Review by fyodorio — CHANGES_REQUESTED** [review]
@AidanB446 looks cool, thanks for digging into it 👍 Now we can talk with ourselves forever! 😅

I have a couple of suggestions to improve the merging dialog, and also there is a small issue with merging itself — the joint shuld better be separated by a new line. Please take a look at the comments.

File: chat.js (hunk 1 above)
Comment (AidanB446) [reply chat.js]: Yes sir, I just updated this for the Y and N options. It definitely makes a lot more sense.

File: chat.js (hunk 2 above)
Comment (AidanB446) [reply chat.js]: I switched it to the second suggested change. Works well now !!

File: chat.js (hunk 3 above)
Comment (AidanB446) [reply chat.js:69]: I will figure this out, and the soultion to it, and get back to you. Thank you for finding the error, and for working with me on this.

Comment (AidanB446) [reply chat.js:69]: Hello @fyodorio

Screenshot from 2024-03-15 09-19-07

For some reason, the line breaks work fine for me. But, a possible solution to the problem is to initially push a new line into chatLog when we type Y for continue

`chatLog.push("  \n");`

This is a change I have implemented. I hope it works, thank you!!

**Review by AidanB446 — COMMENTED** [review]
Hello! Thank you for taking the time to review the PR

I made changes to the lowercase Y and N. And made Y not case sensitive. As for the issue with line breaks, when Y is choosen, it automatically pushes a line break to chatLog which should solve the issue. After testing it out it works for me 
Screenshot from 2024-03-15 09-19-07

Thank you for your time!!

**Review by fyodorio — APPROVED** [review]
@AidanB446 looks awesome, thanks again 🤝

I have renamed the file with respect to [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) format (highly recommended), and added a small fix for a couple of typos (check out [this commit](https://github.com/fyodorio/chat-with-yourself/pull/2/commits/1c3dde452099eec8633f194d2c66f316bf76db52) — my editor also did some [prettier](https://prettier.io/) job there against my will 😅 I hope you don't mind that).

All in all, great job, cheers, merging it!

**Pull request merged**